pkg/webhook/stats_reporter.go

```diff
@@ -53,16 +63,26 @@ func newStatsReporter() (StatsReporter, error) {
 }
 
 // Captures req count metric, recording the count and the duration
-func (r *reporter) ReportAdmissionRequest(response admissionReqRes, d time.Duration) error {
+func (r *reporter) ReportAdmissionRequest(response requestResponse, d time.Duration) error {
+	return r.reportRequest(response, admissionStatusKey, responseTimeInSecM.M(d.Seconds()))
+}
+
+// Captures req count metric, recording the count and the duration
+func (r *reporter) ReportMutationRequest(response requestResponse, d time.Duration) error {
+	return r.reportRequest(response, mutationStatusKey, mutationResponseTimeInSecM.M(d.Seconds()))
+}
+
+// TODO: should this be exported, and ReportAdmissionRequest and ReportMutationRequest removed?
```

Comment:
+1 on exporting this function

Reply:
Done

Reply:
IMO we should have left this as two different functions. My reasoning from a different comment:

> We should collapse these into two functions:
>
> `ReportValidationRequest`
> 
> and
> 
> `ReportMutationRequest`
> 
> That way we can do the fanout into two metrics internally in the metrics interface. That will make a future removal of the deprecated metric easier.

Reply:
Moved back to having different functions for each.
Note that we have 3 functions now (once deprected)

Reply:
Can `ReportValidationRequest()` increment both the new and the deprecated metric? That would avoid code duplication.

Then once we deprecate the old metric we only need to remove the one line.